src/App.js

```diff
@@ -10,8 +10,13 @@ export default function App() {
 
   console.log({ data });
 
-  // Write code here.
-  //
+  useEffect(() => {
+    if(dataType) {
+      fetch(`https://swapi.dev/api/${dataType}/`)
+      .then((response) => response.json())
+      .then((jsonData) => setData(jsonData))
```

Comment:
There's a neat little shortcut trick you could apply here. If you provide any function to something that accepts a function, it will always pass the argument into it. Since you don't have anything special going on here, you could do

`.then(setData)`

The .then will just call your `setData` function and give it what the .then is returning

Reply:
Thank you, I'll keep it in mind!